src/components/limited-restoration-panel/LimitedRestorationPanel.vue

```diff
@@ -0,0 +1,104 @@
+        :disabled="!customMonths" @change="onMonthsChanged" filled />
+    </v-row>
+  </v-radio-group>
+</template>
+<script lang="ts">
+
+import { Component, Emit, Mixins, Prop, Watch } from 'vue-property-decorator'
+import { DateMixin } from '@/mixins' // NB: local mixin (StoryBook can't find it otherwise)
+
+@Component({})
+export default class LimitedRestorationPanel extends Mixins(DateMixin) {
+  @Prop({ default: '' }) readonly currentDate!: string
+
+  @Prop({ default: '' }) readonly expiryDate!: string
+
+  // Local properties
+  private customMonths = false
+  private selectMonths = ''
+  private numberOfMonths = 1
+  private monthRules: ((data: string) => void)[] = []
+
+  /**
+   * Called when the component is mounted.
+   * Auto select the previous months input of user (if any).
+   * If there is no previous input, auto select 24 months.
+   */
+  mounted (): void {
+    this.setMonthRules()
+    if (this.expiryDate !== null && this.expiryDate !== undefined) { // Unit tests fail without the first condition.
+      let draftMonths = this.subtractDates(this.currentDate, this.expiryDate)
+      if (draftMonths !== 24 && draftMonths !== 18 && draftMonths !== 12 && draftMonths !== 6) {
+        this.selectMonths = 'customMonths'
+        this.numberOfMonths = draftMonths
+        this.setCustomMonths(this.numberOfMonths)
+      } else {
+        this.selectMonths = draftMonths
+      }
+    } else {
+      this.selectMonths = 24
+    }
+  }
+
+  /** The validation rules for the Month. */
+  private setMonthRules (): void {
+    this.monthRules = [
+      (v:string) => !!v || 'Please enter the number of months',
+      (v:string) => (v > 0 && v <= 24) || 'This field must be between 1 to 24', // Keep month between 1 and 24
+      (v:string) => (v % 1 === 0) || 'Please enter a valid value' // Check if month is an integer
+    ]
+  }
+
+  // Emit the expiry date.
+  @Emit('expiry')
+  private expiryChanged (event:string): string {
+    return event
+  }
+
+  /**
+   * Set restoration expiry date by adding the custom number of months (1-24) input by user to current date.
+   * Prevent the user from selecting values above 24, lower than 1, and decimals.
+   */
+  onMonthsChanged (): void {
+    if (this.numberOfMonths > 24) {
+      this.numberOfMonths = 24
+    } else if (this.numberOfMonths < 1) {
+      this.numberOfMonths = 1
+    } else if (this.numberOfMonths % 1 !== 0) {
+      this.numberOfMonths = Math.floor(this.numberOfMonths)
+    }
+    this.expiryChanged(this.addMonthsToDate(this.numberOfMonths, this.currentDate))
+  }
+
+  /**
+   * Set if custom number of months is selected.
+   * Set expiry date by adding the value of the selected radio buttons to current date.
+   * @param val The value of the selected radio button
+   */
+   @Watch('selectMonths')
+  private setCustomMonths (val) {
+    this.customMonths = val === 'customMonths'
+    if (val !== 'customMonths') {
```

Comment:
Instead of doing the comparison twice (lines 90 and 91), you could check here whether `customMonths` is False

Reply:
Sev I'm going to keep this as is. But now instead of initializing customMonths as a boolean, it's going to be a string.

Reply:
ok